pycbc/events/stat.py

```diff
@@ -1240,12 +1247,14 @@ def reassign_rate(self, ifo):
         ifo: str
             The ifo to consider.
         """
-        coeff_file = self.files[ifo+'-fit_coeffs']
-        analysis_time = float(coeff_file.attrs['analysis_time'])
+        with h5py.File(self.files[f'{ifo}-fit_coeffs'], 'r') as coeff_file:
+            analysis_time = float(coeff_file.attrs['analysis_time'])
+            fbt = 'fit_by_template' in coeff_file
+
         self.fits_by_tid[ifo]['smoothed_rate_above_thresh'] /= analysis_time
         self.fits_by_tid[ifo]['smoothed_rate_in_template'] /= analysis_time
         # The by-template fits may have been stored in the smoothed fits file
-        if 'fit_by_template' in coeff_file:
+        if fbt:
             self.fits_by_tid[ifo]['fit_by_rate_above_thresh'] /= analysis_time
```

Comment:
For my edification, where are these by-template fit values used?

Reply:
They are used in the cuts; cutting on fit_by values fixed a significant problem in the singles search with isolated templates having lots of triggers, and then that being smoothed over nearby templates and rates are then underestimated

Reply:
So https://github.com/gwastro/pycbc/blob/master/pycbc/events/cuts.py#L44 and https://github.com/gwastro/pycbc/blob/master/pycbc/events/cuts.py#L206